enterprise/core-edge/src/main/java/org/neo4j/coreedge/edge/EdgeStartupProcess.java

```diff
@@ -76,7 +77,47 @@ public void init() throws Throwable
     @Override
     public void start() throws Throwable
     {
-        MemberId source = findCoreMemberToCopyFrom();
+        long retryInterval = 5_000;
+        int attempts = 0;
+        while ( attempts++ < 5 )
+        {
+            MemberId source = findCoreMemberToCopyFrom();
+            try
+            {
+                tryToStart( source );
+                return;
+            }
+            catch ( StoreCopyFailedException e )
+            {
+                log.info( "Attempt #%d to start edge server failed while copying store files from %s.", attempts,
+                        source );
+            }
+            catch ( StreamingTransactionsFailedException e )
+            {
+                log.info( "Attempt #%d to start edge server failed while streaming transactions from %s.", attempts,
+                        source );
+            }
+            catch ( StoreIdDownloadFailedException e )
+            {
+                log.info( "Attempt #%d to start edge server failed while getting store id from %s.", attempts, source );
+            }
+
+            try
+            {
+                Thread.sleep( retryInterval );
+                retryInterval = retryInterval * 2;
```

Comment:
Maybe it's worth capping this value to something like 30'' or 1' so future changes in the try count do not end up with extreme wait values.